JS/homescript.js

```diff
@@ -0,0 +1,83 @@
+const searchBtn = document.getElementById("button-addon2");
+searchBtn.addEventListener("click", getStocks);
+const stockList = document.getElementById("search-resultslist");
+const spinner = document.getElementById("spinner");
+const marquee = document.getElementById("marquee");
+let stocksArray = [];
+
+async function getStocks() {
+  try {
+    stockList.innerHTML = "";
+    spinner.classList.remove("d-none");
+    stocksArray = [];
+    const searchValue = document.getElementById("search-input").value;
+    const fullURL = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/search?query=${searchValue}&limit=10&exchange=NASDAQ`;
+    const response = await fetch(fullURL);
+    const result = await response.json();
+    getSingleStocks(result);
+  } catch {
+    console.log("Error");
+  }
+}
+
+async function getSingleStocks(stocks) {
+  try {
+    for (let currentStock of stocks) {
+      const urlCompanyProfile = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/company/profile/${currentStock.symbol}`;
+      const response = await fetch(urlCompanyProfile);
```

Comment:
when you use the await keyword inside a loop, your http requests will be run synchronously, which means the entire action will take a very long time. You should use Promise.all() in order to send all the needed HTTP requests in the same time.

Reply:
Hi @tomerp20 , I've tried to implement this and I have to submit my milestone up to 7 complete by tomorrow night and I can't get it working. I would appreciate if you could teach me?